docs/DeveloperGuide.md

```diff
@@ -251,77 +270,377 @@ _{Explain here how the data archiving feature will be implemented}_
+2.  SociaLite shows a list of contacts
+3.  User requests to delete a category tagged to a specific contact in the list
+4.  SociaLite deletes the tag
+
+    Use case ends.
+
+*Extensions*
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. SociaLite shows an error message.
+
+      Use case resumes at step 2.
+
+* 3b. The given category does not exist.
+
+    * 3a1. SociaLite shows an error message.
+
+      Use case resumes at step 2.
+
+<br/>
+
+**Use Case 10: Retrieve last queried contact**
+
+*MSS*
+
+1.  User requests to retrieve the information of the last queried contact
+2.  SociaLite returns the entry that fits this criteria
+
+    Use case ends.
+
+
+<br/>
+
+***Customization Tools***
+
+**Use Case 11: Add remark for a specific contact**
+
+*MSS*
+
+1.  User requests to **list contacts (UC02)** or **find contact (UC04)**
+2.  SociaLite returns a list of contacts according to the UC called
+3.  User specifies a remark to be added to the specified contact's entry
+4.  SociaLite adds a remark to the specified contact's entry
+
+    Use case ends.
+
+*Extensions*
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. SociaLite shows an error message.
+
+      Use case resumes at step 2.
+
+<br/>
+
+**Use Case 12: View contact card**
+
+*MSS*
+
+1.  User requests to **list contacts (UC02)** or **find contact (UC04)**
+2.  SociaLite returns a list of contacts according to the UC called
+3.  User specifies the index of his desired contact
+4.  SociaLite opens a contact card which presents all previously stored details associated with the contact
+
+    Use case ends.
+
+*Extensions*
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is invalid.
+
+    * 3a1. SociaLite shows an error message.
+
+      Use case resumes at step 2.
+
+<br/>
+
+***Help Guide & Exiting***
+
+**Use Case 13: View User Guide**
+
+*MSS*
+
+1.  User requests to view User Guide
+2.  SociaLite displays a link to User Guide
+
+    Use case ends.
+
+<br/>
+
+**Use Case 14: Purge contacts stored in SociaLite**
+
+*MSS*
+
+1.  User requests to clear all existing contacts (from demo version)
+2.  SociaLite clears specimen data from memory
+    
+    Use case ends.
+
+<br/>
+
+**Use Case 15: Exit application**
+
+*MSS*
+
+1.  User types in command to exit application
+2.  SociaLite closes
+
+    Use case ends.
+
+
+
 *{More to be added}*
 
-### Non-Functional Requirements
+<br/>
+
+
+
+### Non-Functional Requirements <a name="nfr"></a>
 
 1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
 2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
 3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
```

Comment:
Maybe we can add a point on users who are not so tech savvy, that they should be able to easily learn the basic commands and operate the app with minimal hassle? (Not sure if that's a NFR)

Reply:
Thanks for the suggestion! I've added your suggested NFR in my latest commit.